src/gui/guiButton.cpp

```diff
@@ -819,7 +800,30 @@ void GUIButton::setFromStyle(const StyleSpec& style)
 	} else {
 		setImage(nullptr);
 	}
+
 	BgMiddle = style.getRect(StyleSpec::BGIMG_MIDDLE, BgMiddle);
+
+	// Child padding and offset
+	Padding = style.getRect(StyleSpec::PADDING, BgMiddle);
+
+	GUISkin* skin = dynamic_cast<GUISkin*>(Environment->getSkin());
+	core::vector2d<s32> defaultPressOffset(
+			skin->getSize(irr::gui::EGDS_BUTTON_PRESSED_IMAGE_OFFSET_X),
+			skin->getSize(irr::gui::EGDS_BUTTON_PRESSED_IMAGE_OFFSET_Y));
+	ContentOffset = style.getVector2i(StyleSpec::CONTENT_OFFSET, isPressed()
+			? defaultPressOffset
+			: core::vector2d<s32>(0));
+
+	core::rect<s32> childBounds(
+				Padding.UpperLeftCorner.X + ContentOffset.X,
+				Padding.UpperLeftCorner.Y + ContentOffset.Y,
+				AbsoluteRect.getWidth() + Padding.LowerRightCorner.X + ContentOffset.X,
+				AbsoluteRect.getHeight() + Padding.LowerRightCorner.Y + ContentOffset.Y);
+
+	for(IGUIElement *child : getChildren())

```

Comment:
sorry, but mistake in code style here

Reply:
You're referring to the brace on that for loop, right? If not, please re-open and explain in more detail :)

Reply:
Err, what about a space, i.e. `for (`, not `for(`

Reply:
Good catch!